### PR TITLE
Fix autoplay button wrongly being active after a Ctrl-Enter play

### PR DESCRIPTION
**Pull request by FreezyLemon**
The multiplier and everything were correct before, just the button was still active.

### CODE REVIEW

File: osu.Game/Screens/Select/PlaySongSelect.cs [hunk 1]
```diff
@@ -100,6 +100,7 @@ protected override void OnResuming(Screen last)
             {
                 var autoType = Ruleset.Value.CreateInstance().GetAutoplayMod().GetType();
                 SelectedMods.Value = SelectedMods.Value.Where(m => m.GetType() != autoType).ToArray();
```
Comment (Aergwyn) [inline osu.Game/Screens/Select/PlaySongSelect.cs]: Wouldn't this line be unnecessary if you deselect right afterwards anyways?

**Comment by peppy** [comment]
I'd like to figure out *why* the button was in the wrong state, because this is basically a local fix. If something elsewhere in the game modifies the mods `Bindable` directly the same thing is going to happen.

**Comment by FreezyLemon** [comment]
Note that this is only a bug when _removing_ the mod. Adding the mod works fine (observe by pressing Ctrl-Enter with the mod selection still open. you can see the button being selected properly).

This might have been an oversight, but the current code only tries to select new mods without deselecting the old ones:
Image

(where `SelectTypes` does not deselect any buttons)

**Comment by peppy** [comment]
Yeah, that's the part which should be fixed. You can probably add a fix for that in this PR as well.

**Comment by FreezyLemon** [comment]
Looking at this a bit more, there's some inconsistencies I'd like to address too:
1. There's `SelectTypes` and `DeselectTypes` even though the `SelectTypes` method takes the mod instances, and not their types, as an argument. I'd like to make this more consistent. Should the methods take types, or mod instances as arguments?
2. Should `SelectTypes` maintain its functionality (just add new mods), or should it set (add + remove) the buttons to some 'selection state' for a lack of a better term?

**Review by peppy — APPROVED** [review]

**Pull request merged**